app/models/committees/agenda_type.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+class AgendaType < ApplicationRecord
+  # search
+  include PgSearch
+  pg_search_scope(
+    :search,
+    against: :name,
+    using: { tsearch: { prefix: true } }
+  )
+
+  # reloations
+  has_many :agendas, dependent: :nullify
```

Comment:
`dependent:nullify` kullanıyorsan bu senaryoyu kodun diğer yerlerinde de işletmen lazım. Örneğin aşağıda view'larda `agenda.agenda_type.name` satırı var. Eğer bir agenda_type silinirse, agenda'nın agenda_type'ı bulunamadığı için bu sayfa erişilemez olur, hataya düşer.